core/lib/presentation/extensions/media_type_extension.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:core/data/constants/constant.dart';
+import 'package:core/domain/preview/document_uti.dart';
+import 'package:core/domain/preview/supported_preview_file_types.dart';
+import 'package:core/presentation/resources/image_paths.dart';
+import 'package:http_parser/http_parser.dart';
+
+extension MediaTypeExtension on MediaType {
+  bool isAndroidSupportedPreview() => SupportedPreviewFileTypes.androidSupportedTypes.contains(mimeType);
+
+  bool isIOSSupportedPreview() => SupportedPreviewFileTypes.iOSSupportedTypes.containsKey(mimeType);
+
+  bool isImageFile() => SupportedPreviewFileTypes.imageMimeTypes.contains(mimeType);
+
+  bool isDocFile() => SupportedPreviewFileTypes.docMimeTypes.contains(mimeType);
+
+  bool isPowerPointFile() => SupportedPreviewFileTypes.pptMimeTypes.contains(mimeType);
+
+  bool isExcelFile() => SupportedPreviewFileTypes.xlsMimeTypes.contains(mimeType);
+
+  bool isZipFile() => SupportedPreviewFileTypes.zipMimeTypes.contains(mimeType);
+
+  bool isRtfFile() => SupportedPreviewFileTypes.rtfMimeTypes.contains(mimeType);
+
+  DocumentUti getDocumentUti() => DocumentUti(SupportedPreviewFileTypes.iOSSupportedTypes[mimeType]);
+
+  String getIcon(ImagePaths imagePaths, {String? fileName}) {
+    if (isPDFFile(fileName: fileName) == true || isRtfFile()) {
```

Comment:
rtf should not display as a PDF?

Reply:
We need the correct icon for the `RTF ` file. Please contact the designer to help me.